### PR TITLE
[ Typecasting]  Added the function text to agtype

### PR DESCRIPTION
**Pull request by Zahid07**
Added the function to cast text to agtype
Corrected the catalog test case

Useage:
`SELECT * FROM text_to_agtype('test');`

```
Output:
 text_to_agtype
----------------
 "test"
(1 row)
```

### CODE REVIEW

File: age--1.1.0.sql [hunk 1]
```diff
@@ -3198,6 +3198,14 @@ AS 'MODULE_PATHNAME';
...
 WITH FUNCTION ag_catalog.agtype_to_text(agtype);
 
+CREATE FUNCTION ag_catalog.text_to_agtype(text)
+RETURNS agtype
+LANGUAGE c
+IMMUTABLE
+RETURNS NULL ON NULL INPUT
+PARALLEL SAFE
+AS 'MODULE_PATHNAME';
+
```
Comment (JoshInnis) [inline age--1.1.0.sql:3208]: You now need to add the CREATE CAST command which allows text to be cast like so `SELECT '"This is a string"'::agtype::text;` Also you need to add regression tests

Comment (Zahid07) [reply age--1.1.0.sql:3208]: Okay, I'll add these

Comment (Zahid07) [reply age--1.1.0.sql:3208]: Added the changes

File: regress/sql/expr.sql [hunk 2]
```diff
@@ -2461,6 +2461,17 @@ SELECT * from cypher('list', $$MATCH (u) RETURN labels(u), u$$) as (Labels agtyp
 SELECT * from cypher('list', $$RETURN labels(NULL)$$) as (Labels agtype);
 -- should return an error
 SELECT * from cypher('list', $$RETURN labels("string")$$) as (Labels agtype);
+
+--
+-- text to agtype typecasting
+--
+
+SELECT text_to_agtype('test');
+SELECT '"This is a string"'::agtype::text;
```
Comment (JoshInnis) [inline regress/sql/expr.sql]: This is testing agtype_to_text

Comment (Zahid07) [reply regress/sql/expr.sql]: Please review

**Pull request merged**